### ISSUE TITLE
NAC pool on Cisco 9200 (4 switch in stack)

### ISSUE THREAD

**Issue by kruczek8989** [librenms/librenms#15933]
### The problem

Problem with polling the nac module for Cisco 9200 switches connected in a stack (4 switches).

The problem occurs when polling more than 77 ports (Cisco 9200 switches in the stack have port_id's from 1-233).

The merit is that the parameters from the switch MIB - CISCO-AUTH-FRAMEWORK-MIB they return correct values from the switch, but with debug poller nac I get this error

```
Error polling nac module for xxx.yyy.zzz.vvv. ErrorException: Undefined array key "cafSessionTimeout" in /opt/librenms/LibreNMS/OS/Shared/Cisco.php:405
Stack trace:
#0 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#2 /opt/librenms/app/Providers/ErrorReportingProvider.php(199): call_user_func()
#3 /opt/librenms/LibreNMS/OS/Shared/Cisco.php(405): App\Providers\ErrorReportingProvider->handleError()
#4 /opt/librenms/LibreNMS/Modules/Nac.php(82): LibreNMS\OS\Shared\Cisco->pollNac()
#5 /opt/librenms/LibreNMS/Poller.php(176): LibreNMS\Modules\Nac->poll()
#6 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#7 /opt/librenms/app/Console/Commands/DevicePoll.php(45): LibreNMS\Poller->poll()
#8 /opt/librems/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#9 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#11 /opt/librems/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#12 /opt/librems/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#13 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call()
#14 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#15 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\Component\Console\Command\Command->run()
#16 /opt/librenms/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run()
#17 /opt/librenms/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#18 /opt/librenms/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#19 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#20 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#21 /opt/librenms/lnms(38): App\Console\Kernel->handle()
#22 {main}
```

### Output of ./validate.php

```txt
===========================================
Component | Version
--------- | -------
LibreNMS  | 24.3.0-9-g7dd82a58b (2024-04-10T13:31:34+02:00)
DB Schema | 2024_02_07_151845_custom_map_additions (290)
PHP       | 8.2.13
Python    | 3.8.10
Database  | MariaDB 10.3.38-MariaDB-0ubuntu0.20.04.1
RRDTool   | 1.7.2
SNMP      | 5.8
===========================================

[OK]    Composer Version: 2.4.1
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQL and PHP time match
[OK]    Distributed Polling setting is enabled globally
[OK]    Connected to rrdcached
[OK]    Active pollers found
[OK]    Dispatcher Service is enabled
[OK]    Locks are functional
[OK]    Python wrapper cron entry is not present
[OK]    Redis is functional
[INFO]  fping FAILURES can be ignored if running LibreNMS in a jail without ::1. You may want to test it manually: fping ::1
[FAIL]  /usr/bin/fping6 could not be executed. /usr/bin/fping6 must have CAP_NET_RAW capability (getcap) or suid. Selinux exclusions may be required.
 (/usr/bin/fping6: can't create socket (must run as root?))
[OK]    rrdtool version ok
[OK]    Connected to rrdcached
```

### What was the last working version of LibreNMS?

24.3.0

### Anything in the logs that might be useful for us?

```txt
Error polling nac module for xxx.yyy.zzz.vvv. ErrorException: Undefined array key "cafSessionAuthUserName" in /opt/librenms/LibreNMS/OS/Shared/Cisco.php:400
Stack trace:
#0 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#2 /opt/librenms/app/Providers/ErrorReportingProvider.php(199): call_user_func()
#3 /opt/librenms/LibreNMS/OS/Shared/Cisco.php(400): App\Providers\ErrorReportingProvider->handleError()
#4 /opt/librenms/LibreNMS/Modules/Nac.php(82): LibreNMS\OS\Shared\Cisco->pollNac()
#5 /opt/librenms/LibreNMS/Poller.php(176): LibreNMS\Modules\Nac->poll()
#6 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#7 /opt/librenms/app/Console/Commands/DevicePoll.php(45): LibreNMS\Poller->poll()
#8 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#9 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#11 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#12 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#13 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call()
#14 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#15 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\Component\Console\Command\Command->run()
#16 /opt/librenms/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run()
#17 /opt/librenms/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#18 /opt/librenms/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#19 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#20 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#21 /opt/librenms/lnms(38): App\Console\Kernel->handle()
#22 {main}  
Undefined array key "cafSessionAuthUserName" {"exception":"[object] (ErrorException(code: 0): Undefined array key \"cafSessionAuthUserName\" at /opt/librenms/LibreNMS/OS/Shared/Cisco.php:400)"}
```

**Comment by kruczek8989** [comment]
That's working, now on the stack is active 50 ports and no error.

**Comment by PipoCanaja** [comment]
It will be auto-closed when https://github.com/librenms/librenms/pull/15934 is merged, so nothing to do :) 
Bye

**Comment by kruczek8989** [comment]
> It will be auto-closed when #15934 is merged, so nothing to do :) Bye

Ok, thanks ;)
